src/com/projectkorra/projectkorra/earthbending/lava/LavaSurge.java

```diff
@@ -236,15 +237,15 @@ public void progress() {
 				x = (this.random.nextBoolean()) ? -x : x;
 				z = (this.random.nextBoolean()) ? -z : z;
 
-				fbs.setVelocity(this.direction.clone().add(new Vector(x, 0.2, z)).multiply(1.2));
+				GeneralMethods.setVelocity(this,fbs, this.direction.clone().add(new Vector(x, 0.2, z)).multiply(1.2));
 				fbs.setDropItem(false);
 
 				for (final Block b : this.fracture) {
 					if (this.random.nextBoolean() && b != this.sourceBlock) {
 						final FallingBlock fb = GeneralMethods.spawnFallingBlock(b.getLocation().add(new Vector(0, 1, 0)), Material.MAGMA_BLOCK, Material.MAGMA_BLOCK.createBlockData());
 						ALL_FALLING_BLOCKS.add(fb);
 						this.fallingBlocks.add(fb);
-						fb.setVelocity(this.direction.clone().add(new Vector(this.random.nextDouble() / 10, 0.1, this.random.nextDouble() / 10)).multiply(1.2));
+						GeneralMethods.setVelocity(this,fb, this.direction.clone().add(new Vector(this.random.nextDouble() / 10, 0.1, this.random.nextDouble() / 10)).multiply(1.2));

```

Comment:
Please change to
`GeneralMethods.setVelocity(this, fb, this.direction.clone().add(new Vector(this.random.nextDouble() / 10, 0.1, this.random.nextDouble() / 10)).multiply(1.2));`